src/tests/Interop/NativeLibrary/API/NativeLibraryTests.cs

```diff
@@ -365,6 +370,21 @@ static TestResult TryGetLibraryExport(IntPtr handle, string name)
         });
     }
 
+    static TestResult GetEntryPointModuleHandle()
+    {
+        CurrentTest = nameof(GetEntryPointModuleHandle);
+        return Run(() => {
+            IntPtr moduleHandle = NativeLibrary.GetEntryPointModuleHandle();
+            // Get a well-known export name from corerun to validate that we have a valid handle.
+            bool success = NativeLibrary.TryGetExport(moduleHandle, "GetCurrentClrDetails", out IntPtr address);
+            if (!success)
+                return TestResult.ReturnFailure;
+            if (address == IntPtr.Zero)
+                return TestResult.ReturnNull;
+            return TestResult.Success;
+        });
```

Comment:
Also validate whether or not this handles symbols in non-exe modules? I believe the current implementation will return symbols in non-exe modules on Unix.